### PR TITLE
Go 1.22

### LINKED ISSUE #113: Go 1.22 upgrade check list
See https://github.com/nspcc-dev/neo-go/issues/3310

### PR DESCRIPTION
**Pull request by End-rey**
Closes #113.

Is it also necessary to change here the ubuntu version to the latest in workflows?

### CODE REVIEW

File: pkg/pool/pool.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (p *Pool) recheck(ctx context.Context) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	for i := 0; i < len(p.endpoints); i++ {
+	for i := range len(p.endpoints) {
```
Comment (roman-khimov) [inline pkg/pool/pool.go:106]: Can be `for i:= range p.endpoints`

**Review by roman-khimov — COMMENTED** [review]
Please update dependencies as well. Ubuntu is optional, it's not yet set in stone (https://github.com/nspcc-dev/.github/issues/32)

File: pkg/pool/pool.go (hunk 1 above)
Comment (End-rey) [reply pkg/pool/pool.go:106]: I think about it, but why is it initially written this way?

Comment (roman-khimov) [reply pkg/pool/pool.go:106]: We can ask @smallhive, he's the author of d146024d.

Comment (smallhive) [reply pkg/pool/pool.go:106]: I saw it that way at that moment

**Review by smallhive — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**